tritium-metrics/src/main/java/com/palantir/tritium/metrics/InstrumentedSslEngine.java

```diff
@@ -277,6 +278,23 @@ public String toString() {
         return getClass().getSimpleName() + "{name=" + name + ", delegate=" + engine + '}';
     }
 
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other instanceof InstrumentedSslEngine) {
```

Comment:
curious why instanceof vs `.getClass()` check used above (noting that `InstrumentedSslEngine` is not final while `InstrumentedSslContext` is final)

Reply:
error-prone complained about getClass in a non-final equals method. I considered suppressing because this type is effectively sealed, but opted to avoid suppressions when this works as well.
If `other` uses the java9 implementation, the delegate engine equality check will fail, the type of our InstrumentedSslEngine depends directly on the fields available on the delegate instance.

Reply:
this should be fine, I just don't recall why we didn't make `InstrumentedSslEngine` final as well.

Reply:
I had thought I'd documented this but apparently not! Added a comment explaining the design here: #673